src/github.com/matrix-org/go-neb/services/rssbot/rssbot.go

```diff
@@ -327,8 +376,14 @@ func (s *Service) newItems(feedURL string, allItems []*gofeed.Item) (items []gof
 		//   This will inevitably break for some people, but that group of people are probably smaller, so *shrug*.
 		i.Title = html.UnescapeString(i.Title)
 		i.Description = html.UnescapeString(i.Description)
+		if i.Author != nil {
+			i.Author.Name = html.UnescapeString(i.Author.Name)
+			i.Author.Email = html.UnescapeString(i.Author.Email)
+		}
 
-		items = append(items, *i)
+		if !itemFiltered(i, &mustInclude, &mustNotInclude) {
```

Comment:
Why are you passing pointers to this function? It seems odd since we don't do it anywhere else. You can even just inline the whole lot and do:
```
if !itemFiltered(i, s.Feeds[feedURL].MustInclude, s.Feeds[feedURL].MustNotInclude) {
    ...
}
```

which reduces the number of local vars hanging around.

Reply:
To avoid copying the two `includeRules` structs each time, though they're not that big. As for the vars, I declare them to avoid doing a map lookup for each item.c

Reply:
SGTM